payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -152,19 +179,21 @@ internal class PaymentSheetViewModel @Inject internal constructor(
         }
     }
 
-    @VisibleForTesting
-    fun fetchIsGooglePayReady() {
-        if (isGooglePayReady.value == null) {
-            if (args.isGooglePayEnabled) {
-                viewModelScope.launch {
-                    val isGooglePayReady = withContext(workContext) {
-                        googlePayRepository.isReady().first()
+    fun setupGooglePay(activity: ComponentActivity) {
```

Comment:
Please try to avoid using `ComponentActivity` in `ViewModel`, probably pass the lifecycleScope and activityResultLauncher here, this will also make them easier to mock

Reply:
Done